tests/src/python/test_qgsvectorfilewriter.py

```diff
@@ -93,6 +93,46 @@ def testWrite(self):
 
         writeShape(self.mMemoryLayer, 'writetest.shp')
 
+    def testWriteWithBoolField(self):
+
+        # init connection string
+        dbconn = 'dbname=\'qgis_test\''
+        if 'QGIS_PGTEST_DB' in os.environ:
+            dbconn = os.environ['QGIS_PGTEST_DB']
+
+        # create a vector layer
+        vl = QgsVectorLayer('{} table="qgis_test"."boolean_table" sql='.format(dbconn), "testbool", "postgres")
```

Comment:
A bit overkill to use a postgres source for that. You might just create a dummy 

test.csv:
id,myfield
1,1

test.csvt:
Integer,Integer(Boolean)

Reply:
> A bit overkill to use a postgres source for that.

Actually, it was to stick as closely as possible to the underlying ticket.